test/parallel/test-http-1.0.js

```diff
@@ -34,26 +34,21 @@ function test(handler, request_generator, response_validator) {
   let server_response = '';
 
   server.listen(0);
-  server.on('listening', function() {
-    const c = net.createConnection(this.address().port);
+  server.on('listening', common.mustCall(() => {
+    const c = net.createConnection(server.address().port);
 
     c.setEncoding('utf8');
 
-    c.on('connect', function() {
-      c.write(request_generator());
-    });
+    c.on('connect', common.mustCall(() => c.write(request_generator())));
+    c.on('data', common.mustCall((chunk) => server_response += chunk));
```

Comment:
I agree with that point. But aren't we at the mercy of different platforms and whatnot for the flow of data? If it can be done reliably, and without platform specific checks, then 👍